Source/WebCore/page/Navigation.cpp

```diff
@@ -700,8 +700,9 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
         RefPtr fromNavigationHistoryEntry = currentEntry();
         ASSERT(fromNavigationHistoryEntry);
 
-        // FIXME: Create finished promise
-        m_transition = NavigationTransition::create(navigationType, *fromNavigationHistoryEntry);
+        auto& domGlobalObject = *jsCast<JSDOMGlobalObject*>(scriptExecutionContext()->globalObject());
+        JSC::JSLockHolder locker(domGlobalObject.vm());
```

Comment:
Does this need to still be in scope for the updateURLAndHistory call?  I think it only needs to be in scope for the DeferredPromise::create call.  Could you use { and } to add a scope for this?